[zuoshihua] iP

zuoshihua opened the pull request:
### Tundra

> "Done is better than perfect." - Sheryl Sandberg ([source](https://www.goodreads.com/quotes/6486999-done-is-better-than-perfect-i-have-tried-to-embrace))

Tundra is simple yet powerful task tracker. It is,

- text-based
- easy to learn
- ~FAST~ *BLAZINGLY* FAST to use

Try it now by,

1. Download the latest version [here](https://github.com/zuoshihua/ip/releases/download/A-Jar/tundra.jar)
2. Double-click it
3. Add your tasks
4. Sit back and let it take care of the rest :rocket:

Best of all? It's **FREE**

Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

---

If you're a Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
public static void main(String[] args) {
    new Tundra("./data/tundra.txt").run();
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

zuoshihua committed:
Rename main class

zuoshihua committed:
Change Duke to Tundra

zuoshihua committed:
Add greeting

zuoshihua committed:
Added support for echo

zuoshihua committed:
Added support for create and retrieve operations

zuoshihua committed:
Created Task class with fields for name and completed

zuoshihua committed:
Added setter for Task field completed

zuoshihua committed:
Added missing print statement and code to set/unset Task

zuoshihua committed:
Created Task subclasses Deadline, Event & Todo

zuoshihua committed:
New field by and custom toString for Deadline

zuoshihua committed:
New fields from & to, custom toString for Event

zuoshihua committed:
Custom toString for Todo

zuoshihua committed:
Add switch statement for different types of Task and their respective string parsing

zuoshihua committed:
Change Duke to Tundra

zuoshihua committed:
Add sample input to test create, retrieve and update ops

zuoshihua committed:
Add expected output for comparison

zuoshihua committed:
Add custom TundraException

zuoshihua committed:
Add error handling code that provide helpful hints

zuoshihua committed:
Add custom format for exception message

zuoshihua committed:
Add delete command to remove from ArrayList

zuoshihua committed:
Add delete command to list of commands

zuoshihua committed:
Update README

zuoshihua committed:
Add enum for Task symbol

zuoshihua committed:
Update toString to use TaskEnum

zuoshihua committed:
Add Database class to handle persistence

zuoshihua committed:
remove Task enum

zuoshihua committed:
update Task subclasses not to rely on Task enum

zuoshihua committed:
Add new method to write a Task in human readable format

zuoshihua committed:
Add subclass specific method to wrtie it in human readable format

zuoshihua committed:
Add methods to retrieve and safe tasks on filesystem

zuoshihua committed:
Update main method to utilise Database

zuoshihua committed:
Merge pull request #1 from zuoshihua/branch-Level-7

Branch level 7

zuoshihua committed:
Change date and time fields from type String to LocalDateTime

zuoshihua committed:
Add code to parse date and time from user input

zuoshihua committed:
Add code to format and parse dates and times

zuoshihua committed:
Merge pull request #2 from zuoshihua/branch-Level-8

Branch level 8

zuoshihua committed:
Organize classes into packages

zuoshihua committed:
Add method fromStoredString to initialize each Task from saved file

zuoshihua committed:
Add util Storage to handle loading and saving tasks to filesystem

zuoshihua committed:
Remove Database class since deprecated

zuoshihua committed:
Add Storage class to replace Database class

zuoshihua committed:
Rename Task subclasses for more clarity

zuoshihua committed:
Add toString to return task symbol

zuoshihua committed:
Update stringify methods to get symbol from Task enum

zuoshihua committed:
Update load and save methods

zuoshihua committed:
Add command to list tasks

zuoshihua committed:
Update TundraException to use Ui for templating

zuoshihua committed:
Add Ui method to read user input

zuoshihua committed:
Move date time formats to Parser

zuoshihua committed:
Replace Tundra class

zuoshihua committed:
Ensure Task classes use datetime formatter from Parser

zuoshihua committed:
Make Task an abstract class

zuoshihua committed:
Add all commands into enum

zuoshihua committed:
Create a class for each command

zuoshihua committed:
Add TaskList class to perform operations on tasks collection

zuoshihua committed:
Update Storage class to return status integer

zuoshihua committed:
Update Parser to convert user input into command

zuoshihua committed:
Update Ui to only have instance methods

zuoshihua committed:
Merge pull request #3 from zuoshihua/branch-A-MoreOOP

Branch a more oop

zuoshihua committed:
Move Command classes to models package

zuoshihua committed:
Move Command classes to views package

zuoshihua committed:
Merge pull request #4 from zuoshihua/branch-A-Packages

Branch a packages

zuoshihua committed:
Merge branch 'add-gradle-support'

zuoshihua committed:
Update build.gradle to point to Tundra

zuoshihua committed:
Add unit tests for Parser and DeadlineTask

zuoshihua committed:
Merge pull request #5 from zuoshihua/branch-A-JUnit

Add unit tests for Parser and DeadlineTask

zuoshihua committed:
Add Javadoc

zuoshihua committed:
Format code according to standard

zuoshihua committed:
Remove println statement

zuoshihua committed:
Add new command to search tasks by description

zuoshihua committed:
Include new find command in command enum

zuoshihua committed:
Merge pull request #6 from zuoshihua/branch-A-JavaDoc

Add Javadoc

zuoshihua committed:
Merge branch 'master' into branch-A-CodingStandard

zuoshihua committed:
Merge pull request #7 from zuoshihua/branch-A-CodingStandard

Format code according to standard

zuoshihua committed:
Merge pull request #8 from zuoshihua/branch-Level-9

Branch level 9

SanShaoQian commented on src/main/java/tundra/Tundra.java:
```diff
@@ -0,0 +1,54 @@
+/**
+ * Represents the entry point of the application
+ */
+public class Tundra {
```
Good job having meaningful names for your classes! 👍

SanShaoQian commented on src/main/java/tundra/models/Task.java:
```diff
@@ -0,0 +1,68 @@
+public class Task {
+
+    private String name;
+    private boolean completed;
```
perhaps this can be renamed to `isCompleted` to make it sound more like a boolean variable?

SanShaoQian commented on src/main/java/tundra/views/ListCommand.java:
```diff
@@ -0,0 +1,40 @@
+     * A task's string representation differs from their stored string representation.
+     */
+    @Override
+    public void execute(TaskList taskList, Ui ui, Storage storage) {
```
Good job following the coding standard for the naming of methods! 👍

SanShaoQian commented on src/test/java/tundra/models/DeadlineTaskTest.java:
```diff
@@ -0,0 +1,23 @@
+public class DeadlineTaskTest {
+
+    @Test
+    public void toStoredString_datetimeGiven_correctFormatReturned() {
```
Perhaps consider a more concise name for this method?

zuoshihua committed:
Add Checkstyle config

zuoshihua committed:
Configure Gradle to use Checkstyle

zuoshihua committed:
Ignore Eclipse files

zuoshihua committed:
Move Checkstyle config to correct location

zuoshihua committed:
Ensure Gradle uses Java 17 and Azul SDK

zuoshihua committed:
Fix Checkstyle violations

zuoshihua committed:
Merge pull request #9 from zuoshihua/branch-A-Checkstyle

Branch a checkstyle

zuoshihua committed:
Add FXML for main chat and chat message

zuoshihua committed:
Update Ui class to cache latest message

zuoshihua committed:
Update Tundra class to hand over command output

zuoshihua committed:
Add Main and App class

zuoshihua committed:
Close scanner after use

zuoshihua committed:
Add Theme enum for light and dark themes

zuoshihua committed:
Add controller for main chat and chat message components

zuoshihua committed:
Add stylesheets for both light and dark themes

zuoshihua committed:
Update to point to Main class which is the entrypoint

zuoshihua committed:
Appease Checkstyle

popova-mariia commented on src/main/java/tundra/exceptions/TundraException.java:
```diff
@@ -0,0 +1,11 @@
+package tundra.exceptions;
```
Maybe you could consider different subtypes of tundra exception for different situations (like unrecognised command ect)?

popova-mariia commented on src/main/java/tundra/models/EventTask.java:
```diff
@@ -0,0 +1,77 @@
+    /**
+     * Creates a new <code>EventTask</code> with the specified name, the
+     * specified date, the specified starting date and time,
+     * the specified ending date and time and the completed flag unset.
```
Shouldnt there be an empty line between description and @pagam? I noticed that issue in several other places too.

popova-mariia commented on src/main/java/tundra/models/Task.java:
```diff
@@ -0,0 +1,68 @@
+        try {
+            setCompleted(parts[0].equalsIgnoreCase("1"));
+            setName(parts[1]);
+        } catch (ArrayIndexOutOfBoundsException e) {
```
Maybe you could consider using if else statements instead of try and catch, since exceptions should not control the flow of program but rather address unexpected situations?

popova-mariia commented on src/main/java/tundra/models/TaskEnum.java:
```diff
@@ -0,0 +1,24 @@
+package tundra.models;
```
I like that u did Enum given that it was optional.

popova-mariia commented on src/main/java/tundra/views/ByeCommand.java:
```diff
@@ -0,0 +1,30 @@
+
+
+/**
+ * Represents the command "bye".
```
Maybe you could consider explaining more on what the command is supposed to do (eg bye command terminates the program and saves created tasks ect)? could be applied to other classes as well.

popova-mariia reviewed (commented):
Overall great job 😊

zuoshihua committed:
Update styles to make text more legible

zuoshihua committed:
Merge pull request #10 from zuoshihua/branch-Level-10

Branch level 10

zuoshihua committed:
Update Ui to accept array of strings

zuoshihua committed:
Update Ui to accept variable arguments

zuoshihua committed:
Merge pull request #11 from zuoshihua/branch-A-Varargs

Branch a varargs

zuoshihua committed:
Set chat message component height to negative infinity

zuoshihua committed:
Delete gradle.properties

zuoshihua committed:
Update bye command to gracefully shutdown application

zuoshihua committed:
Refactor bye command to use PauseTransition for delay

zuoshihua committed:
Enable assertions

Add enableAssertions to build.gradle as instructed by this post on
StackOverflow
https://stackoverflow.com/questions/48396274/how-to-enable-assertions-in-the-gradle-run-task

zuoshihua committed:
Add assertions for non-nullable objects

zuoshihua committed:
Extract adding of task to a private method

zuoshihua committed:
Merge pull request #13 from zuoshihua/branch-A-Assertions

Branch a assertions

zuoshihua committed:
Merge branch 'master' into branch-A-CodeQuality

zuoshihua committed:
Merge pull request #14 from zuoshihua/branch-A-CodeQuality

Branch a code quality

zuoshihua committed:
Add equals method

zuoshihua committed:
Add contains method

zuoshihua committed:
Add force command to set the forced flag

zuoshihua committed:
Add forced flag

zuoshihua committed:
Add check if task already exists, add task only if forced flag is set

zuoshihua committed:
Merge pull request #15 from zuoshihua/C-DetectDuplicates

C detect duplicates

zuoshihua committed:
Add a brief user guide

zuoshihua committed:
Appease checkstyle

zuoshihua committed:
Set window title to Tundra

zuoshihua committed:
Include JavaFX dependencies in fat JAR

zuoshihua committed:
Rephrase JavaDoc

zuoshihua committed:
Merge pull request #16 from zuoshihua/branch-A-UserGuide

Branch a user guide

zuoshihua committed:
Correct typo in unmark command
